src/openms/include/OpenMS/METADATA/ProteinModificationSummary.h

```diff
@@ -0,0 +1,69 @@
+// --------------------------------------------------------------------------
+//                   OpenMS -- Open-Source Mass Spectrometry
+// --------------------------------------------------------------------------
+// Copyright The OpenMS Team -- Eberhard Karls University Tuebingen,
+// ETH Zurich, and Freie Universitaet Berlin 2002-2020.
+//
+// This software is released under a three-clause BSD license:
+//  * Redistributions of source code must retain the above copyright
+//    notice, this list of conditions and the following disclaimer.
+//  * Redistributions in binary form must reproduce the above copyright
+//    notice, this list of conditions and the following disclaimer in the
+//    documentation and/or other materials provided with the distribution.
+//  * Neither the name of any author or any participating institution
+//    may be used to endorse or promote products derived from this software
+//    without specific prior written permission.
+// For a full list of authors, refer to the file AUTHORS.
+// --------------------------------------------------------------------------
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL ANY OF THE AUTHORS OR THE CONTRIBUTING
+// INSTITUTIONS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
+// OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+// WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR
+// OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
+// ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// --------------------------------------------------------------------------
+// $Maintainer: Timo Sachsenberg $
+// $Authors: $
+// --------------------------------------------------------------------------
+
+#pragma once
+
+#include <map>
+
+#include <OpenMS/METADATA/MetaInfoInterface.h>
+#include <OpenMS/CHEMISTRY/ResidueModification.h>
+
+namespace OpenMS
+{
+  /**
+   * @brief Maps a protein position to all observed modifications and associated statistics 
+   * 
+   * For example, to store that position 10 maps to Oxidation (M) which was observed in 123 PSMs.
+   * 
+   */
+  struct OPENMS_DLLAPI ProteinModificationSummary
+  {
+    /// basic modification statistic
+    struct OPENMS_DLLAPI Statistics : public MetaInfoInterface
```

Comment:
do you really really need the base class here? It makes for a hidden interface which noones knows what is actually stored in there...

Reply:
I guess most may want to store false localization rates, probabilities etc.
what would be a better option? storing a vector of cv params as a member?

Reply:
explicit members with 'NaN' defaults.
I think memory consumption should not be a problem here, given that the number of proteins is not too large to pose a problem.

Reply:
One problem I see is that storing those extra "scores" in a file might get more difficult. Wouldn't we need a new format every time we add a new score here?

Reply:
how you store that in a file is a different matter :)